src/openrct2/ride/coaster/BolligerMabillardTrack.hpp

```diff
@@ -11819,6 +11819,438 @@ void bolliger_mabillard_track_booster(
+                    break;
+                case 1:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27434, 0, 6, 32, 20, 3, height + 8);
+                    break;
+                case 2:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27431, 0, 6, 32, 20, 3, height + 8);
+                    break;
+                case 3:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27437, 0, 6, 32, 20, 3, height + 8);
+                    break;
+            }
+            paint_util_set_segment_support_height(
+                session,
+                paint_util_rotate_segments(
+                    SEGMENT_BC | SEGMENT_C0 | SEGMENT_C4 | SEGMENT_CC | SEGMENT_D0 | SEGMENT_D4, direction),
+                0xFFFF, 0);
+            paint_util_set_general_support_height(session, height + 32, 0x20);
+            break;
+        case 2:
+            switch (direction)
+            {
+                case 0:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27429, 0, 6, 32, 20, 3, height - 5);
+                    break;
+                case 1:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27435, 0, 6, 32, 20, 3, height - 5);
+                    break;
+                case 2:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27430, 0, 6, 32, 20, 3, height - 5);
+                    break;
+                case 3:
+                    PaintAddImageAsParentRotated(
+                        session, direction, session.TrackColours[SCHEME_TRACK] | 27436, 0, 6, 32, 20, 3, height - 5);
+                    break;
+            }
+            metal_a_supports_paint_setup(
+                session, METAL_SUPPORTS_TUBES_INVERTED, 4, 0, height - 5, session.TrackColours[SCHEME_SUPPORTS]);
+            switch (direction)
+            {
+                case 1:
+                    paint_util_push_tunnel_right(session, height, TUNNEL_SQUARE_FLAT);
+                    break;
+                case 2:
+                    paint_util_push_tunnel_left(session, height, TUNNEL_SQUARE_FLAT);
+                    break;
+            }
+            paint_util_set_segment_support_height(
+                session,
+                paint_util_rotate_segments(
+                    SEGMENT_BC | SEGMENT_C0 | SEGMENT_C4 | SEGMENT_CC | SEGMENT_D0 | SEGMENT_D4, direction),
+                0xFFFF, 0);
+            paint_util_set_general_support_height(session, height + 32, 0x20);
+            break;
+    }
+}
+
+inline void bolliger_mabillard_track_right_twist_up_to_down(
```

Comment:
This is not yet correct. All other functions use a template where the support type is passed, while you have hardcoded it `METAL_SUPPORTS_TUBES_INVERTED`. Also, please use `static` instead of `inline` like the other functions do - inline does not seem to make the binary smaller at this point, and we would do the whole file in one go anyway.

Reply:
The game doesn't build when using `static`, that was the whole reason I had to use `inline` in the firstplace :P 

And how should I approach fixing the metal supports to be non-hardcoded? 
By using a line similar to `metal_a_supports_paint_setup(session, supportType, 2, 2, height, session.TrackColours[SCHEME_SUPPORTS]);` ?

Reply:
No, just look at the signature of the other functions. It will include a template parameter called `supportType` which is used inside the function.

Also, `static` works for all other functions. If it doesn’t work for yours, you must have made a mistake somewhere.

Reply:
It's really odd, I just changed the `METAL_SUPPORTS_TUBES_INVERTED` to `supportType` and now it's erroring out on me again. Again bouncing back an error from the FlyingRC. 
`C:\Users\karst\Documents\GitHub\OpenRCT2\src\openrct2\ride\coaster\BolligerMabillardTrack.hpp(11849,26): error C2065: 'supportType': undeclared identifier (compiling source file ride\coaster\FlyingRollerCoaster.cpp) [C:\Users\karst\Docum
ents\GitHub\OpenRCT2\src\openrct2\libopenrct2.vcxproj]`

I have elaborated a bit over here: https://discord.com/channels/264137540670324737/691735032531779684/930233773008752640